commcare_connect/utils/middleware.py

```diff
@@ -40,3 +42,29 @@ def __call__(self, request):
     def process_view(self, request, view_func, view_args, view_kwargs):
         if hasattr(view_func, "cls") and view_func.cls.versioning_class is not None:
             request.include_version_headers = True
+
+
+class SentryContextMiddleware:
+    """Add details to Sentry context.
+    Should be placed after '"commcare_connect.users.middleware.OrganizationMiddleware",'
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+        try:
+            from sentry_sdk import Scope  # noqa: F401
+        except ImportError:
+            raise MiddlewareNotUsed
+
+        if not getattr(settings, "SENTRY_DSN", None):
+            raise MiddlewareNotUsed
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        from sentry_sdk import configure_scope
+
+        with configure_scope() as scope:
```

Comment:
Why use this pattern instead of the one shown in [the docs](https://docs.sentry.io/platforms/python/enriching-events/scopes/) using `get_current_scope`? Is there a difference between them. I can't actually find this one in the docs.

Reply:
On further look, it looks like this function is deprecated https://docs.sentry.io/platforms/python/migration/1.x-to-2.x#custom-instrumentation

Reply:
I have updated it to use set_user and set_tag. (Previously I adapted this from HQ which was using 2.8 )